src/c++/perf_analyzer/docs/examples/profile.py

```diff
@@ -175,73 +223,134 @@ def collect_periodic_latencies(args, filename):
     return bins
 
 
-def calculate_avg_periodic_latencies(args, filename):
+def calculate_avg_periodic_latencies(args, profile_result, filename):
     """Calculate average token-to-token latency for each request period."""
     bins = collect_periodic_latencies(args, filename)
 
     latencies = []
     for bin in bins:
-        latencies.append(np.mean(bin) / 1_000_000_000)
-    return latencies
+        latencies.append(np.mean(bin) / 1_000_000)
+
+    profile_result.avg_periodic_t2t_latencies = latencies
 
 
 def collect_latencies(requests):
     # Example json demonstrating format:
     #   see client/src/c++/perf_analyzer/docs/examples/decoupled_output_file.json
     first_token_latencies = []
+    generation_latencies = []
     token_to_token_latencies = []
     requests = requests["experiments"][0]["requests"]
+    for r in requests:
+        init_request, responses = r["timestamp"], r["response_timestamps"]
+        first_token_latencies.append((responses[0] - init_request) / 1_000_000)
+        generation_latencies.append((responses[-1] - responses[0]) / 1_000_000)
+        token_to_token_latencies = []
+        for prev_res, res in pairwise(responses):
+            token_to_token_latencies.append((res - prev_res) / 1_000_000)
+    return first_token_latencies, generation_latencies, token_to_token_latencies
+
+
+def calculate_online_metrics(args, profile_result, filename):
+    """Calculate online metrics for more fine-grained performance information."""
+    if not args.stream:
+        return  # skip if offline
+
+    requests = load_json_data(filename)
+    latencies = collect_latencies(requests)
+    first_token_latencies, generation_latencies, token_to_token_latencies = latencies
+
+    profile_result.avg_first_token_latency = np.mean(first_token_latencies)
+    profile_result.avg_total_t2t_latency = np.mean(token_to_token_latencies)
+
+    profile_result.max_first_token_latency = max(first_token_latencies)
+    profile_result.min_first_token_latency = min(first_token_latencies)
+    profile_result.avg_first_token_latency = np.mean(first_token_latencies)
```

Comment:
This was already set here:

https://github.com/triton-inference-server/client/pull/431/files#diff-cd92f27dd4906cc0da4a3ee0731bf185f9745e1df6c284f26813a225635af728R263

Reply:
Good catch. Fixed.